dev-tools/validation/check-authz-route-matrix.py

```
@@ -61,64 +491,2043 @@ def string_list(value: Any, field: str, errors: list[str]) -> list[str]:
+            cardinality_errors,
+        )
+        if route is not None:
+            validate_idempotency_contract(
+                route,
+                f"{service} {route_name}",
+                errors,
+                required_fields_cache,
+            )
+
 
+def validate_refresh_roles_routes(
```

Ruff (0.16.0)

[warning] Too many branches (15 > 12)

(PLR0912)